### ISSUE TITLE
Header Cache broken?

### ISSUE THREAD

**Issue by patrickbenkoetter** [neomutt/neomutt.github.io#126]
Could it be the recent release NeoMutt 20241114 broke the header cache? Eversince then I noted mutt to build a list of mailbox headers everytime I start NeoMutt and access my mailboxes.

My current config sets this:
```

# Caching
set header_cache="~/.mutt/cache/"
set tmpdir="~/.mutt/tmp/"
set message_cachedir="~/.mutt/messages/"
set message_cache_clean = yes
```

This is my release:

```
NeoMutt 20241114

System: Darwin 24.1.0 (arm64)
ncurses: ncurses 6.5.20240427 (compiled with 6.5.20240427)
libiconv: 1.17
libidn2: 2.3.7 (compiled with 2.3.7)
GPGME: 1.24.0
OpenSSL: OpenSSL 3.4.0 22 Oct 2024
libnotmuch: 5.6.0
PCRE2: 10.44 2024-06-07
storage: tokyocabinet, lmdb
compression: zlib

Configure options: --sysconfdir=/opt/homebrew/etc --autocrypt --gss --disable-idn --idn2 --lmdb --nls --notmuch --pcre2 --sasl --sqlite --tokyocabinet --zlib --with-idn2=/opt/homebrew/opt/libidn2 --with-lua=/opt/homebrew/opt/lua --with-ncurses=/opt/homebrew/opt/ncurses --with-ssl=/opt/homebrew/opt/openssl@3 --with-sqlite=/opt/homebrew/opt/sqlite --with-iconv=/opt/homebrew/opt/libiconv --disable-debug --disable-dependency-tracking --prefix=/opt/homebrew/Cellar/neomutt/20241114 --libdir=/opt/homebrew/Cellar/neomutt/20241114/lib

Compilation CFLAGS: -std=c11 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/opt/homebrew/Cellar/gpgme/1.24.0_1/include -I/opt/homebrew/Cellar/libassuan/3.0.1/include -I/opt/homebrew/opt/libgpg-error/include -I/opt/homebrew/Cellar/neomutt/20241114/include -I/opt/homebrew/include/lua -I/opt/homebrew/Cellar/neomutt/20241114/include -I/opt/homebrew/Cellar/neomutt/20241114/include -I/opt/homebrew/opt/sqlite/include -I/opt/homebrew/Cellar/neomutt/20241114/include -DNCURSES_WIDECHAR -I/opt/homebrew/opt/ncurses/include -I/opt/homebrew/opt/libiconv/include -I/opt/homebrew/Cellar/openssl@3/3.4.0/include -I/opt/homebrew/Cellar/libidn2/2.3.7/include -I/opt/homebrew/Cellar/pcre2/10.44/include -I/opt/homebrew/opt/lmdb/include -I/opt/homebrew/Cellar/tokyo-cabinet/1.4.48/include -I/opt/homebrew/Cellar/neomutt/20241114/include -O2

Einstellungen bei der Kompilierung:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo -gnutls +gpgme
  -gsasl +gss +hcache -homespool +idn -inotify -locales_hack +lua +nls +notmuch
  +openssl +pcre2 +pgp +sasl +smime +sqlite +truecolor

MAILPATH="/var/mail"
PKGDATADIR="/opt/homebrew/Cellar/neomutt/20241114/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/opt/homebrew/etc"

```

**Comment by patrickbenkoetter** [comment]
No need to fix it. Everything is okay. Must have been a problem on my end. Sorry for the noise.

**Comment by flatcap** [comment]
No worries.
Some day, I'll write a doc about the header cache and new releases.

tl;dr
The header cache is closely bound to the code.
Development often means a new release invalidates the old cache -- causing temporary slowdowns.
